gix-dir/tests/walk/mod.rs

```diff
@@ -4276,3 +4276,161 @@ fn type_mismatch_ignore_case_clash_file_is_dir() {
          If there was no special handling for this, it would have found the file (`d` in the index, icase), which would have been wrong."
     );
 }
+
+#[test]
+fn top_level_slash_with_negations() -> crate::Result {
+    for repo_name in ["slash-in-root-and-negated", "star-in-root-and-negated"] {
+        let root = fixture(repo_name);
+        let ((out, _root), entries) = collect(&root, None, |keep, ctx| walk(&root, ctx, options_emit_all(), keep));
+        assert_eq!(
+            out,
+            walk::Outcome {
+                read_dir_calls: 2,
+                returned_entries: entries.len(),
+                seen_entries: 5,
+            }
+        );
+        assert_eq!(
+            entries,
+            &[
+                entry_nokind(".git", Pruned).with_property(DotGit).with_match(Always),
```

Comment:
Before the fix, this entry was ignored which was part of the problem. With the fix, it turned into a pruned `DotGit` entry, which naturally won't be touched by `gix clean`.